### ISSUE TITLE
Important features no longer labelled when zooming in

### ISSUE THREAD

**Issue by thany** [organicmaps/organicmaps#4048]
An important feature might be the main train station, primary points of interest, the city name, or neighbourhood names.

These are all unlabelled when zooming in far enough, which makes it hard to get my bearings in a large-ish city. Go over to Matsuyama, Ehime, Japan. You'll see the city name when zoom out. Zoom in a little, and it disappears. All the while, where is Matsuyama station? Where are other important stations, like a central tram station and terminal stations? The whole city becomes a jumble of unnamed roads (roads are largely unnamed in Japan and cannot be used for navigation).

Even when zooming in to Dogo Onsen (good luck finding it...) the terminal tram station is not displayed until zooming all the way in - but then I've already found it, haven't I 😀

The bottom line is that the app probably isn't correctly prioritising what to display at certain zoom levels. Now I don't know if it comes down to data, or algorithms, and if this is different by country, but the fact is that Google Maps has absolutely nailed this, so please copypaste what they are doing 😉 (easier said than done, I know that).

**Comment by biodranik** [comment]
Let's formalize it in a list: which objects should have higher drawing priorities on which zoom level.
You can turn on zoom levels by typing `?debug-info` in the search

**Comment by thany** [comment]
Is that for me to decide? I can give it a go, but I am by no means a mapping expert...

1. From scale=9 I would expect central train station for large cities. How large depends on which cities are on the map, I guess.
2. From scale=10 I would expect central train stations like Matsuyama, Utrecht Central, Berlin.
3. From scale=11 the city name disappears while zooming in. After zooming in it's still at scale=11 though, so I'm guessing it's rounded.
4. From scale=12 I would still expect to see the city name, and central/hub/terminal stations. Tram stations and important bus stations should appear.
5. From scale=14 I would expect tram stations to appear (maybe at scale=13 already, not sure). Currently, tram stations are drawn at scale=17, which is too "late".
6. At scale=12 it manages to draw Yuzuki Castle, which is a ruin that is known as Dogo Park (probably a data issue) but more importantly it doesn't draw Matsuyama Castle, which should probably be drawn at scale=10 already. It's one of the important castles in the country and people are undoubtedly looking for it. Same for others of course.

Again, this might vary from place to place. This is just an isolated example. I'm not sure how you'd formalise these differences.

Perhaps we have more mapping experts in our midst who can share some ideas on how to approach this.

**Comment by Jean-BaptisteC** [comment]
@thany can you check on latest version, we have doing a big rework on style priorities?

**Comment by thany** [comment]
It's a little bit better. But the city name still disappears at zoom level 12, and gets replaced by a myriad of neighbourhopod names, which I feel are in too large a font. While not unimportant, I'm not sure they are important at this scale just yet.

Regardless, the main train station is still not labelled until I zoomed in all the way to level 15, at which point I don't really need the label anymore. I'd be nice if I were able to see where I'm zooming into, so to speak. I say 15, but it only appears at the far end of zoom level 15. When just going from 14 to 15, it doesn't display yet. It only displays kind of like halfway into zoom level 15.

I think the underlying problem is if you "hard code" what type of features become visible (and stop being visible) at certain zoom levels, you will always have problems such as the ones I described earlier. Bigger features might need to be displayed & labelled at a smaller zoom level than smaller features *of the same type*.

Now I'm not sure if bigger cities can be labelled "earlier" than small cities. Same for train stations, neighbourhoods, and all kinds of other features. I feel the crux is: you cannot indiscriminately tie a zoom level to a feature type.

I tested this in Matsuyama, Ehime, Japan.
